spec.proto

```diff
@@ -0,0 +1,55 @@
+/**
+ * CloudEvent Protobuf Format
+ *
+ * - Required context attributes are explicity represented.
+ * - Optional and Extension context attributes are carried in a map structure.
+ * - Data may be represented as binary, text, or protobuf messages.
+ */
+
+syntax = "proto3";
+
+package io.cloudevents.v1;
+
+import "google/protobuf/any.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_package = "io.cloudevents.v1.proto";
+
+message CloudEvent {
+
+  // -- CloudEvent Context Attributes
+
+  // Required
+  string id = 1;
+  string source = 2; // URI-reference
+  string spec_version = 3;
+  string type = 4;
+
+  // Optional & Extension
+  map<string, CloudEventAttribute> attributes = 5;
+
+  // -- CloudEvent Data (Bytes, Text, or Proto)
+  oneof data_oneof {
+    bytes binary_data = 6;
+    string text_data = 7;
+    google.protobuf.Any proto_data = 8;
+  }
+
+  /**
+   * The CloudEvent specification defines
+   * seven atrribute types...
+   */
+
+  message CloudEventAttribute {
+
+    oneof attr_oneof {
+      bool cloud_event_boolean = 1;
+      sfixed32 cloud_event_integer = 2;
+      string cloud_event_string = 3;
+      bytes cloud_event_binary = 4;
+      string cloud_event_uri = 5;
+      string cloud_event_uri_reference = 6;
+      google.protobuf.Timestamp cloud_event_timestamp = 7;
```

Comment:
I'd remove the `cloud_event_` prefix from these fields -- since it's inside a CloudEvent, I worry that you'll get ReallyLongNames.With.ReallyLongStutters.

Reply:
Ack